satpy/dataset/dataid.py

```diff
@@ -344,15 +344,19 @@ def from_dict(self, keyvals):
         return self.__class__(self._id_keys, **keyvals)
 
     @classmethod
-    def from_dataarray(cls, array, default_keys=minimal_default_keys_config):
+    def from_dataarray(cls, array, default_keys=None):
         """Get the DataID using the dataarray attributes."""
+        if default_keys is None:
+            default_keys = minimal_default_keys_config
```

Comment:
I don't agree here. The default value will only ever be read, and the only reason this is a dict is because there is not builtin frozen dict.